src/abstractdataframe/abstractdataframe.jl

```diff
@@ -762,13 +762,20 @@ function completecases(df::AbstractDataFrame, col::Colon=:)
     end
     res = trues(size(df, 1))
     for i in 1:size(df, 2)
-        res .&= .!ismissing.(df[!, i])
+        if Missing <: eltype(df[!, i])
```

Comment:
It is better to extract `df[!, i]` once to a variable as it is not guaranteed that compiler will optimize this out (though it should in theory). Same comment below.